### DISCUSSION TITLE
Put an icon into the tab

### DISCUSSION THREAD

**Discussion by Taldrit78** [fluentribbon/Fluent.Ribbon#1173] [Q&A] [answered]
I wanted to put an icon into the tab. The header property is of type object. So I thought it would be the normal WPF way. that I can put a StackPanel inside that contains what I wanted to show. But it seems, that the object is then shown with a simple .ToSting() implementation.
Is that correct?
I am still using version 8. Would it work with version 10? We wanted to upgrade soon, but didn't get the time from out bosses so far.
I looked into the code. There it seems you extended the IHeaderedControl with templates, but only a one or two line header. So there can only be one or two line of text?

EDIT: My thoght was about instead of putting the header into a Textblock, you maybe could check if the content of the HeaderProperty is a UIElement or FrameworkElement and if yes, put it into a ContentPresenter. And if it is a string, put it into a TextBlock. So for noone would anything change, but it would give us more flexibility in what we want to show in the Tab.Header.

**Comment by batzen** [comment, marked as answer]
The default HeaderTemplate for RibbonTabItem is defined as a TextBlock in it's style.
You have to either inherit the style and overwrite the HeaderTemplate there or change it for your specific RibbonTabItem instance.

**Reply by batzen** [reply]
I am not sure if that's also the case for version 8 though.
